ports/raspberrypi/common-hal/busio/UART.c

```diff
@@ -375,30 +257,19 @@ void common_hal_busio_uart_set_timeout(busio_uart_obj_t *self, mp_float_t timeou
 }
 
 uint32_t common_hal_busio_uart_rx_characters_available(busio_uart_obj_t *self) {
-    // This assignment is only here because the usart_async routines take a *const argument.
-    // struct usart_async_descriptor * const usart_desc_p = (struct usart_async_descriptor * const) &self->usart_desc;
-    // struct usart_async_status async_status;
-    // usart_async_get_status(usart_desc_p, &async_status);
-    // return async_status.rxcnt;
-    return 0;
+    return ringbuf_num_filled(&ringbuf[self->uart_id]);
 }
 
 void common_hal_busio_uart_clear_rx_buffer(busio_uart_obj_t *self) {
-    // This assignment is only here because the usart_async routines take a *const argument.
-    // struct usart_async_descriptor * const usart_desc_p = (struct usart_async_descriptor * const) &self->usart_desc;
-    // usart_async_flush_rx_buffer(usart_desc_p);
-
+    // Prevent conflict with uart irq.
+    irq_set_enabled(self->uart_irq_id, false);
+    ringbuf_clear(&ringbuf[self->uart_id]);
```

Comment:
Want to clear the UART FIFO here too?

Reply:
In a continuous stream... won't the FIFO keep filling up while we erase it?

Reply:
You should be able to clear it faster than it is populated. (You are reading bytes at 125 mhz.)